[CI] Multiple DockerTests failures

**Build scan**:
https://gradle-enterprise.elastic.co/s/k52yxd2twpqla

**Reproduces locally?**:
Didn't try

**Applicable branches**:
8.0

**Failure history**:


**Failure excerpt**:
 1> [2021-11-02T07:43:10,092][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test122CanUseDockerLoggingConfig-seed#[AD72A6E41071F0AD]-workerest_thread_info] Ran: [bash, -c, docker rm -f e9e7344a06765472311039d084b5f0e4cc6fd5336e39b84a13a81a064cec0d86] exitCode = [0] stdout = [e9e7344a06765472311039d084b5f0e4cc6fd5336e39b84a13a81a064cec0d86] stderr = []
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
    PID   USER     TIME  COMMAND
        1 elastics  0:00 /bin/tini -- /app/elasticsearch.sh
        8 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      521 elastics  0:23 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Xms4m -Xmx64m -XX:+UseSerialGC -Des.cgroups.hierarchy.override=/ -Des.cgroups.hierarchy.override=/ -Des.path.home=/usr/share/elasticsearch -Des.path.conf=/usr/share/elasticsearch/config -Des.distribution.flavor=default -Des.distribution.type=docker -cp /usr/share/elasticsearch/lib/*:/usr/share/elasticsearch/modules/x-pack-core/*:/usr/share/elasticsearch/modules/x-pack-security/*:/usr/share/elasticsearch/lib/tools/security-cli/* org.elasticsearch.xpack.security.cli.AutoConfigureNode -Eingest.geoip.downloader.enabled=false -Ediscovery.type=single-node
      707 elastics  0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
        at __randomizedtesting.SeedInfo.seed([AD72A6E41071F0AD:725A83B60EBA1377]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.test122CanUseDockerLoggingConfig(DockerTests.java:801)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test122CanUseDockerLoggingConfig

I noticed other test failures related to DockerTests with a similar output on both 8.0 and master. I hesitate to create a dedicated for each failure because I suspect they share the same cause.

----

#### DockerTests.test070BindMountCustomPathConfAndJvmOptions on 8.0
https://gradle-enterprise.elastic.co/s/5r5njmryixqqo
```
1> [2021-11-02T08:17:29,803][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test070BindMountCustomPathConfAndJvmOptions-seed#[49AD1AB120008B8C]-workerest_thread_info] Ran: [bash, -c, docker rm -f b76a8996a84abff7ae001aafc3e06487e5944271c42e7a5e0002e32637f380bb] exitCode = [0] stdout = [b76a8996a84abff7ae001aafc3e06487e5944271c42e7a5e0002e32637f380bb] stderr = [bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)]
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
        PID TTY      STAT   TIME COMMAND
          1 ?        Ss     0:00 /bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper
          8 ?        S      0:00 /bin/bash /usr/share/elasticsearch/bin/elasticsearch
        732 ?        S      0:00 /bin/bash /usr/share/elasticsearch/bin/elasticsearch
        733 ?        Sl     0:00 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -cp /usr/share/elasticsearch/lib/* org.elasticsearch.tools.launchers.JvmOptionsParser /usr/share/elasticsearch/config /usr/share/elasticsearch/plugins
        775 pts/2    Rs+    0:00 ps ax
        778 ?        Sl     0:00 /usr/share/elasticsearch/jdk/bin/java -XX:+UseG1GC -Djava.io.tmpdir=/tmp/elasticsearch-7202926394221363839 -XX:+HeapDumpOnOutOfMemoryError -XX:+ExitOnOutOfMemoryError -XX:HeapDumpPath=data -XX:ErrorFile=logs/hs_err_pid%p.log -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m -Des.cgroups.hierarchy.override=/ -Xms1024m -Xmx1024m -Xshare:off -XX:+PrintFlagsFinal -version

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
    bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
        at __randomizedtesting.SeedInfo.seed([49AD1AB120008B8C:BCBAB38916C4F7D9]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test070BindMountCustomPathConfAndJvmOptions

184 tests completed, 1 failed, 171 skipped
```

#### DockerTests.test160CheckImageHealthcheckDefinition on master
https://gradle-enterprise.elastic.co/s/fmf3xdsf2x5ky

```
1> [2021-11-01T21:05:10,592][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test160CheckImageHealthcheckDefinition-seed#[355F9B9972FC2F1E]-workerest_thread_info] Ran: [bash, -c, docker rm -f 23e38b83e99360ed9064dbf738946690ce72432b8343c882735ef9a4e5e68d38] exitCode = [0] stdout = [23e38b83e99360ed9064dbf738946690ce72432b8343c882735ef9a4e5e68d38] stderr = [bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)]
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
        PID TTY      STAT   TIME COMMAND
          1 ?        Ss     0:00 /bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper
          8 ?        S      0:00 /bin/bash /usr/share/elasticsearch/bin/elasticsearch
        515 ?        Sl     0:22 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Xms4m -Xmx64m -XX:+UseSerialGC -Des.cgroups.hierarchy.override=/ -Des.cgroups.hierarchy.override=/ -Des.path.home=/usr/share/elasticsearch -Des.path.conf=/usr/share/elasticsearch/config -Des.distribution.flavor=default -Des.distribution.type=docker -cp /usr/share/elasticsearch/lib/*:/usr/share/elasticsearch/modules/x-pack-core/*:/usr/share/elasticsearch/modules/x-pack-security/*:/usr/share/elasticsearch/lib/tools/security-cli/* org.elasticsearch.xpack.security.cli.AutoConfigureNode -Eingest.geoip.downloader.enabled=false -Ediscovery.type=single-node
        718 pts/1    Rs+    0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
    bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
        at __randomizedtesting.SeedInfo.seed([355F9B9972FC2F1E:E0EAD92CEBD6890D]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test160CheckImageHealthcheckDefinition

```

#### DockerTests.test050BasicApiTestson 8.0
https://gradle-enterprise.elastic.co/s/rtcufmloig2ky
```
1> [2021-11-01T23:30:46,021][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test050BasicApiTests-seed#[920C398B7F73B843]-workerest_thread_info] Ran: [bash, -c, docker rm -f 433ff9fa26f7765fa732a5d3a194b8c9baf84ef9b73a6a48d1760b94f9da25b1] exitCode = [0] stdout = [433ff9fa26f7765fa732a5d3a194b8c9baf84ef9b73a6a48d1760b94f9da25b1] stderr = []
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
    PID   USER     TIME  COMMAND
        1 elastics  0:00 /bin/tini -- /app/elasticsearch.sh
        8 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      713 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      714 elastics  0:00 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -cp /usr/share/elasticsearch/lib/* org.elasticsearch.tools.launchers.JvmOptionsParser /usr/share/elasticsearch/config /usr/share/elasticsearch/plugins
      757 elastics  0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
        at __randomizedtesting.SeedInfo.seed([920C398B7F73B843:94CF9D5F51605195]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test050BasicApiTests

```

#### DockerTests.test150MachineDependentHeap on 8.0
https://gradle-enterprise.elastic.co/s/otxuhcitkk7tu
```

1> [2021-11-01T23:54:17,548][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test150MachineDependentHeap-seed#[85D87D8435AF907A]-workerest_thread_info] Ran: [bash, -c, docker rm -f c7f95a2319dbc553ebb75f7d83c71355f9921c9857c05fdaaa07c68e33802256] exitCode = [0] stdout = [c7f95a2319dbc553ebb75f7d83c71355f9921c9857c05fdaaa07c68e33802256] stderr = []
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
    PID   USER     TIME  COMMAND
        1 elastics  0:00 /bin/tini -- /app/elasticsearch.sh
        7 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      720 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      721 elastics  0:00 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -cp /usr/share/elasticsearch/lib/* org.elasticsearch.tools.launchers.JvmOptionsParser /usr/share/elasticsearch/config /usr/share/elasticsearch/plugins
      758 elastics  0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
        at __randomizedtesting.SeedInfo.seed([85D87D8435AF907A:4F8D7D94F7F4168A]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test150MachineDependentHeap
 
```

#### DockerTests.test091ElasticsearchShardCliPackaging on 8.0
https://gradle-enterprise.elastic.co/s/rj67bmuzzxyfk
```
1> [2021-11-01T23:44:06,599][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test091ElasticsearchShardCliPackaging-seed#[ECD02AB52BDE9477]-workerest_thread_info] Ran: [bash, -c, docker rm -f caee95a1d1ba1ca8bc94464850f57d9d25004c92668b18bc4ff56dab0276c758] exitCode = [0] stdout = [caee95a1d1ba1ca8bc94464850f57d9d25004c92668b18bc4ff56dab0276c758] stderr = []
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
    PID   USER     TIME  COMMAND
        1 elastics  0:00 /bin/tini -- /app/elasticsearch.sh
        8 elastics  0:00 {elasticsearch} /bin/bash /usr/share/elasticsearch/bin/elasticsearch
      514 elastics  0:22 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Xms4m -Xmx64m -XX:+UseSerialGC -Des.cgroups.hierarchy.override=/ -Des.cgroups.hierarchy.override=/ -Des.path.home=/usr/share/elasticsearch -Des.path.conf=/usr/share/elasticsearch/config -Des.distribution.flavor=default -Des.distribution.type=docker -cp /usr/share/elasticsearch/lib/*:/usr/share/elasticsearch/modules/x-pack-core/*:/usr/share/elasticsearch/modules/x-pack-security/*:/usr/share/elasticsearch/lib/tools/security-cli/* org.elasticsearch.xpack.security.cli.AutoConfigureNode -Eingest.geoip.downloader.enabled=false -Ediscovery.type=single-node
      694 elastics  0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
        at __randomizedtesting.SeedInfo.seed([ECD02AB52BDE9477:8A0341F1FF9339EE]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test091ElasticsearchShardCliPackaging

```

#### DockerTests.test084SymlinkToFileWithInvalidPermissionsIsRejected on master
https://gradle-enterprise.elastic.co/s/rlo26zuhsgb7u

```
1> [2021-11-02T04:43:03,449][INFO ][o.e.p.u.Shell            ] [TEST-DockerTests.test084SymlinkToFileWithInvalidPermissionsIsRejected-seed#[62AF4EA3DAF8B558]-workerest_thread_info] Ran: [bash, -c, docker rm -f 89b7ff414b4a0ebc83ace9ed951e131c1c073198fd9d687424ddb294f0c7bfa9] exitCode = [0] stdout = [89b7ff414b4a0ebc83ace9ed951e131c1c073198fd9d687424ddb294f0c7bfa9] stderr = []
  2> java.lang.AssertionError: Elasticsearch container did not start successfully.

    ps output:
        PID TTY      STAT   TIME COMMAND
          1 ?        Ss     0:00 /bin/tini -- /usr/local/bin/docker-entrypoint.sh eswrapper
          7 ?        S      0:00 /bin/bash /usr/share/elasticsearch/bin/elasticsearch
        528 ?        Sl     0:22 /usr/share/elasticsearch/jdk/bin/java -Xshare:auto -Xms4m -Xmx64m -XX:+UseSerialGC -Des.cgroups.hierarchy.override=/ -Des.cgroups.hierarchy.override=/ -Des.path.home=/usr/share/elasticsearch -Des.path.conf=/usr/share/elasticsearch/config -Des.distribution.flavor=default -Des.distribution.type=docker -cp /usr/share/elasticsearch/lib/*:/usr/share/elasticsearch/modules/x-pack-core/*:/usr/share/elasticsearch/modules/x-pack-security/*:/usr/share/elasticsearch/lib/tools/security-cli/* org.elasticsearch.xpack.security.cli.AutoConfigureNode -Eingest.geoip.downloader.enabled=false -Ediscovery.type=single-node
        730 pts/4    Rs+    0:00 ps ax

    Stdout:
    Created elasticsearch keystore in /usr/share/elasticsearch/config/elasticsearch.keystore

    Stderr:
        at __randomizedtesting.SeedInfo.seed([62AF4EA3DAF8B558:3E75DE8BA2CA71BE]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.elasticsearch.packaging.util.docker.Docker.waitForElasticsearchToStart(Docker.java:177)
        at org.elasticsearch.packaging.util.docker.Docker.runContainer(Docker.java:118)
        at org.elasticsearch.packaging.test.DockerTests.setupTest(DockerTests.java:105)

Tests with failures:
 - org.elasticsearch.packaging.test.DockerTests.test084SymlinkToFileWithInvalidPermissionsIsRejected

```

--- a/qa/os/src/test/java/org/elasticsearch/packaging/util/docker/Docker.java
+++ b/qa/os/src/test/java/org/elasticsearch/packaging/util/docker/Docker.java
@@ -68,7 +68,7 @@ public class Docker {
     public static final Shell sh = new Shell();
     public static final DockerShell dockerShell = new DockerShell();
     public static final int STARTUP_SLEEP_INTERVAL_MILLISECONDS = 1000;
-    public static final int STARTUP_ATTEMPTS_MAX = 20;
+    public static final int STARTUP_ATTEMPTS_MAX = 30;
 
     /**
      * Tracks the currently running Docker image. An earlier implementation used a fixed container name,

